feat(components): [el-dropdown] support custom arrow icon

xiongmao86 opened the pull request:
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

Closes #5671.
- [x] dropdown 增加了一个split-button-icon选项，可以自定义标题右面的箭头标签。
- [x] sub-menu增加自定义箭头标签的选项。

我添加了一个选项split-button-icon，当split-button存在时，可以使用这个自定义选项设置改变右面的箭头标签。

我想要实现的功能是这样的，输入是`<el-dropdown>`则直接显示default slot的值，输入是`<el-dropdown split-button>`显示default slot加上默认箭头符号，输入是`<el-dropdown split-button="CaretBottom">` 显示default slot加上自定义箭头符号CaretBottom。但是我不太熟悉 props的写法，不知道去哪里看或者去哪里问，有没有聊天室可以讨论相关的信息，所以发个pr上来，希望哪位大佬看到了可以帮我一下。

sxzz commented on packages/components/dropdown/src/dropdown.vue:
```diff
@@ -67,7 +67,9 @@
           class="el-dropdown__caret-button"
         >
-          <el-icon class="el-dropdown__icon"><arrow-down /></el-icon>
+          <el-icon class="el-dropdown__icon"
+            ><component :is="splitButtonIcon"
+          /></el-icon>
```
```suggestion
          <el-icon class="el-dropdown__icon">
            <component :is="splitButtonIcon" />
          </el-icon>
```

xiongmao86 replied:
好的，谢谢。

sxzz commented:
现在的写法没啥问题，请调整下格式即可。

JeremyWuuuuu commented:
Looking good, but remember that you need to update the documentation in `docs` for file `dropdown.md`, also you will need to add unit test for this feature. If you don't know how to do that, you may find references by searching those components that provides `icon` as prop

xiongmao86 committed:
feat(components): [dropdown] support custom arrow icon

When split-button is set on a dropdown, you can use split-button-icon to customize arrow icon.

fix #5671

xiongmao86 committed:
feat(components): [menu] customize title icons of all submenus

Provide a sub-menu-icon-down to customize title icons that are pointing down in all submenus. And
sub-menu-icon-right to customize title icons that are pointing right in all submenus.

fix #5671

msidolphin commented on packages/components/menu/src/menu.ts:
```diff
@@ -58,6 +63,14 @@ export const menuProps = buildProps({
     type: Boolean,
     default: true,
   },
+  subMenuIconDown: {
```
I think `subMenuArrowDownIcon` is better, isn't it?

msidolphin commented on packages/components/menu/src/menu.ts:
```diff
@@ -58,6 +63,14 @@ export const menuProps = buildProps({
+    type: definePropType<string | Component>([String, Object]),
+    default: () => ArrowDown,
+  },
+  subMenuIconRight: {
```
`subMenuArrowRightIcon`

msidolphin commented:
Please resolve the conflicts

xiongmao86 commented:
Comments resolved,  please take a look when you have time.

xiongmao86 commented:
`format`, and pass `lint` phase.

xiongmao86 commented:
@sxzz @msidolphin  @JeremyWuuuuu, did you forget my pr, is there anything I can do to get it merged?  Did I do anything wrong?

xiaoxian521 commented:
Hey, thank you very much for your contribution, but here are some conflicts that need to be solved by you. We can continue to review the code after solving. One thing to remind is that if the conflicts is still not resolved after `10` days, sorry, we will close this pr ❤️

xiongmao86 commented:
@xiaoxian521 please take a look when you have time.

xiongmao86 commented:
What is the `Playground Preview` used for? I am not adding example, can I skip it?

xiongmao86 commented:
Build successful, linted and formatted.

xiongmao86 commented:
Do I have to update `dev` and rebase my branch upon it?

xiongmao86 committed:
chore: pnpm format

sxzz commented:
Just wait for reviewing...
